### ISSUE TITLE
quote-props: ["error", "consistent-as-needed", { "numbers": true }]

### ISSUE THREAD

**Issue by rasenplanscher** [eslint/eslint#8093]
**What rule do you want to change?**
`quote-props`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer for some situations, more for others

**How will the change be implemented? (New option, new default behavior, etc.)?**
New object option `numbers` for string option `consistent-as-needed`, consistent with the same object option for string option `as-needed`

**Please provide some example code that this change will affect:**
```js
var object1 = {
    foo: 'bar',
    baz: 42,
    42: true
};
var object2 = {
    'foo': 'bar',
    'baz': 42,
    '42': true
};
```

**What does the rule currently do for this code?**
Warn about unnecessary quotes for `object2`

**What will the rule do after it's changed?**
Warn about missing quotes for `object1`

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. Out of interest, what's the use-case for only quoting numbers?

**Comment by rasenplanscher** [comment]
The use-case is not quoting *only* numbers, but quoting all keys if one or more keys cannot be used with dot-notation, which includes numbers.

Thus, these would be valid:
```js
var object3 = {
    foo: 'bar',
    baz: 42,
    qux: true
};
var object4 = {
    'foo': 'bar',
    'baz': 42,
    'qux-lorem': true
};
```

These would not:
```js
var object3 = {
    foo: 'bar',
    baz: 42,
    '42': true
};
var object4 = {
    foo: 'bar',
    baz: 42,
    'qux-lorem': true
};
```

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
